<div style="text-align: center; background-color: #0A6EBD; font-family: 'Trebuchet MS', Arial, sans-serif; color: white; padding: 20px; font-size: 40px; font-weight: bold; border-radius: 0 0 0 0; box-shadow: 0px 6px 8px rgba(0, 0, 0, 0.2); line-height: 1.5;">
    <center style="color:white;font-size:35px;padding:15px"><b> ĐỒ ÁN CUỐI KỲ </b></center>
<center style="color:black;font-size:30px;padding:15px"><b> LẬP TRÌNH CHO KHOA HỌC DỮ LIỆU</b></center>
</div>


## Import các thư viện cần thiết

In [6]:
import pandas as pd

## 1. Khám phá dữ liệu

### Đọc dữ liệu vào dataframe

In [7]:
df = pd.read_csv('../Data/who_life_exp.csv')

In [8]:
df.head()

country country_code  region  year  life_expect  life_exp60  \
0  Angola          AGO  Africa  2000     47.33730    14.73400   
1  Angola          AGO  Africa  2001     48.19789    14.95963   
2  Angola          AGO  Africa  2002     49.42569    15.20010   
3  Angola          AGO  Africa  2003     50.50266    15.39144   
4  Angola          AGO  Africa  2004     51.52863    15.56860   

   adult_mortality  infant_mort  age1-4mort  alcohol  ...  che_gdp    une_pop  \
0         383.5583     0.137985    0.025695  1.47439  ...  1.90860  16395.473   
1         372.3876     0.133675    0.024500  1.94025  ...  4.48352  16945.753   
2         354.5147     0.128320    0.023260  2.07512  ...  3.32946  17519.417   
3         343.2169     0.122040    0.021925  2.20275  ...  3.54797  18121.479   
4         333.8711     0.115700    0.020545  2.41274  ...  3.96720  18758.145   

   une_infant  une_life  une_hiv  une_gni  une_poverty  une_edu_spend  \
0       122.2    46.522      1.0   2530.0         32.3        2.60753   
1       118.9    47.059      1.1   2630.0          NaN            NaN   
2       115.1    47.702      1.2   3180.0          NaN            NaN   
3       110.8    48.440      1.3   3260.0          NaN            NaN   
4       106.2    49.263      1.3   3560.0          NaN            NaN   

   une_literacy  une_school  
0           NaN         NaN  
1      67.40542         NaN  
2           NaN         NaN  
3           NaN         NaN  
4           NaN         NaN  

[5 rows x 32 columns]

## 2. Tiền xử lí dữ liệu

### Có bao nhiêu dòng và bao nhiêu cột của tập dữ liệu?

In [9]:
num_rows, num_columns = df.shape
print(f'Bộ dữ liệu có {num_rows} dòng và {num_columns} cột.')


Bộ dữ liệu có 3111 dòng và 32 cột.


### Ý nghĩa của từng dòng dữ liệu là gì?
Mỗi dòng thể hiện số liệu giá trị cho từng cột được thu thập được từ WHO, có thể ở một số cột sẽ không có số liệu vì chưa thu thập được. 
Có tổng cộng 3111 dòng số liệu.

### Có dòng nào bị trùng lặp không?

In [10]:
duplicated_rows = df[df.duplicated()]

if duplicated_rows.empty:
    print("Không có dòng trùng lặp.")
else:
    print("Có dòng trùng lặp.")

Không có dòng trùng lặp.


### Ý nghĩa của từng cột dữ liệu là gì?
1. country - Tên của đất nước.
2. country_code - Mã 3 chữ cái xác định của mỗi quốc gia.
3. region - Châu lục.
4. year - Năm dữ liệu được thu thập.
5. life_expect -Tuổi thọ trung bình khi sinh.
6. life_exp60 - Tuổi thọ trung bình ở tuổi 60.
7. adult_mortality - Tỷ lệ tử vong ở người trưởng thành ở cả hai giới (xác suất tử vong từ 15 đến 60 tuổi trên 1000 dân).
8. infant_mort - Tỷ suất tử vong trẻ sơ sinh dưới 1 tuổi.
9. age1-4mort - Tỷ suất tử vong trẻ em từ 1-4 tuổi.
10. alcohol - Mức tiêu thụ rượu bình quân đầu người được ghi nhận ở người lớn (trên 15 tuổi) (lít).
11. bmi -Chỉ số khối cơ thể trung bình (kg/m2), chỉ số dinh dưỡng/mức độ béo phì.
12. age5-19thinness - Tỷ lệ trẻ em từ 5-19 tuổi bị gầy quá mức (Chỉ số suy dinh dưỡng).
13. age5-19obesity - Tỷ lệ trẻ em từ 5-19 tuổi bị béo phì.
14. hepatitis - Tỷ lệ tiêm chủng viêm gan B (HepB) cho trẻ 1 tuổi (%).
15. measles - Tỷ lệ tiêm chủng vắc xin sởi (MCV1) cho trẻ 1 tuổi (%).
16. polio - Tỷ lệ tiêm chủng bại liệt (Pol3) cho trẻ 1 tuổi (%).
17. diphtheria - Tỷ lệ tiêm chủng bạch hầu, uốn ván và ho gà (DTP3) cho trẻ 1 tuổi (%).
18. basic_water - Dân số được tiếp cận nguồn nước sạch.
19. doctors - Số bác sỹ trên 10.000 dân. Chỉ số tài nguyên chăm sóc sức khỏe.
20. hospitals - Số bệnh viện trên 100.000 dân. Chỉ số năng lực y tế.
21. gni_capita - Tổng thu nhập quốc dân bình quân đầu người theo sức mua tương đương (đô la quốc tế).
22. gghe-d: Chi tiêu y tế chính phủ trong nước (GGHE-D) tính theo phần trăm tổng sản phẩm quốc nội (GDP) (%).
23. che_gdp: Chi tiêu y tế hiện tại (CHE) tính theo phần trăm tổng sản phẩm quốc nội (GDP) (%).
24. une_pop: Dân số thất nghiệp ước tính (1.000 người).
25. une_infant: Tỷ lệ tử vong ở trẻ sơ sinh (trên 1.000 trẻ).
26. une_life: Tuổi thọ kỳ vọng khi sinh của người thất nghiệp (năm).
27. une_hiv: Tỷ lệ nhiễm HIV, tổng cộng (% dân số trong độ tuổi 15-49).
28. une_gni: Thu nhập quốc dân (GNI) bình quân đầu người của người thất nghiệp (USD).
29. une_poverty: Tỷ lệ nghèo đầu người ở mức 1,90 USD một ngày (PPP) (% dân số).
30. une_edu_spend: Chi tiêu của chính phủ cho giáo dục tính theo phần trăm GDP (%).
31. une_literacy: Tỷ lệ biết chữ ở người trưởng thành, 15 tuổi trở lên (%).
32. une_school: Số năm đi học trung bình (ISCED 1 trở lên) của người trên 25 tuổi.